src/veileder/kandidatliste/side-header/rekrutteringsstatus/useSletteLagredeStillinger.ts

```diff
@@ -0,0 +1,25 @@
+import { useEffect } from 'react';
+import { KandidatlisteIder } from './useLagreKandidatlisteIder';
+
+const useSletteKandidatlisteIderFraLukkedata = (
+    kandidatlisteId: string,
+    besatteStillinger: number,
+    antallStillinger: number | null,
+    lukkedata: KandidatlisteIder,
+    setLukkedata: (KandidatlisteIder) => void
+) => {
+    useEffect(() => {
+        if (
+            lukkedata &&
+            Array.from(lukkedata).includes(kandidatlisteId) &&
+            antallStillinger &&
+            besatteStillinger < antallStillinger
+        ) {
+            const newSet = new Set(lukkedata);
+            newSet.delete(kandidatlisteId);
+            setLukkedata(newSet);
+        }
+    }, [besatteStillinger, antallStillinger]);
```

Comment:
Vil ikke eslint her klage på at dere har en dependency-array, men bruker andre variabler enn dem som er definert, kandidatlisteId og lukkedata? Trenger dere dependency-arrayen i det hele tatt, kan den fjernes?

Reply:
En god huskeregel er at `useEffect` brukes til å synkronisere props med noe annet, i dette tilfellet Local storage. Deps-arrayen kan brukes hvis det er for tungt, performance-wise, å kjøre useEffect-en på hver render

Står mer her, men den er dritlang https://overreacted.io/a-complete-guide-to-useeffect/

Reply:
Om vi trenge rdependency-arrayen: Nei, tror ikke det, det er bare en ytelsesoptimalisering, i flg. https://reactjs.org/docs/hooks-effect.html#tip-optimizing-performance-by-skipping-effects  I samme tekst (den gule boksen) får jeg inntrykk av at det et vanskelig å angi riktig innhold i dependency-arrayen. Og det står "We recommend using the exhaustive-deps rule as part of our eslint-plugin-react-hooks package. It warns when dependencies are specified incorrectly and suggests a fix." Så nå tenker jeg at vi enten bør droppe denepndency-arrayen eller bruke den anbefalte exhaustive-deps.

Reply:
Ang "Vil ikke eslint her klage på at dere har en dependency-array, men bruker andre variabler enn dem som er definert, kandidatlisteId og lukkedata?": Får ingen klager jeg i IntelliJ...